src/components/ProjectCard/ProjectHoverCard.tsx

```diff
@@ -88,6 +89,23 @@ export const ProjectHoverCard: FC<ProjectCardProps> = ({
     router.push(`/project/${project.slug}`);
   };
 
+  // New token price logic
+  const maxContributionPOLAmountInCurrentRound = 200000 * (10 ^ 18); // Adjust the max cap later from backend
+  const tokenPriceRange = useTokenPriceRange({
+    contributionLimit: maxContributionPOLAmountInCurrentRound,
+    contractAddress: project.abc?.fundingManagerAddress || '',
+  });
+
+  const [POLPrice, setPOLPrice] = useState<number | null>(null);
+
+  useEffect(() => {
+    const fetchPOLPrice = async () => {
+      const price = await fetchTokenPrice(); // Fetch token price in USD
+      setPOLPrice(price);
+    };
+    fetchPOLPrice();
+  }, []);
```

Comment:
This can be replaced by a `useQuery`
Like here https://github.com/GeneralMagicio/abc-launcher/blob/main/src/hooks/usePolTokenPrice.ts